packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -605,6 +617,134 @@ class _AndroidViewState extends State<AndroidView> {
   }
 }
 
+class _WindowsViewState extends State<Win32View> {
+  bool _initialized = false;
+  Win32ViewController? _controller;
+  late FocusNode _node, _before, _after;
+  bool _inner_focus = false;
+  bool _tabbing_out = false; // True only when in the process of moving focus away from the PV with keyboard nav.
+
+  void _initializeOnce() {
+    if (_initialized) {
+      return;
+    }
+    _initialized = true;
+    _createNewWin32View();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _initializeOnce();
+  }
+
+  @override
+  void didUpdateWidget(Win32View oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    if (oldWidget.viewType != widget.viewType) {
+      _controller = null;
+      _createNewWin32View();
+    }
+  }
+
+  Future<void> _createNewWin32View() async {
+    final int id = platformViewsRegistry.getNextPlatformViewId();
+    _node = FocusNode(debugLabel: 'Windows view: $id');
+    _before = FocusNode(debugLabel: 'Capture forward navigation');
+    _after = FocusNode(debugLabel: 'Capture backwards navigation');
+    final Win32ViewController controller = await PlatformViewsService.initWin32View(id: id, viewType: widget.viewType, onFocus: () {
+      _inner_focus = true;
+      _node.requestFocus();
+      print('Node $id stole focus');
+    }, onTabOut: (int reason) {
+      if (reason == 0) {
+        return;
+      }
+      _tabbing_out = true;
+      _inner_focus = true;
+      if (reason == 1) {
+        _after.requestFocus();
+      } else if (reason == 2) {
+        _before.requestFocus();
+      }
+    });
+    setState(() {
+      _controller = controller;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final Win32ViewController? controller = _controller;
+    if (controller == null) {
+      return const SizedBox.expand();
+    }
+    return Semantics(
+      label: 'Platform view placeholder',
+      child: Stack(
+        children: [
```

Comment:
As the focus change callback cannot distinguish between forward, backwards, and non-directional navigation, three focusable widgets are layered in order to achieve this distinction. The first widget will gain focus when navigation is forward-directed, and the last will when it is backwards.